File: backend/src/main/java/com/xxgwy/modules/user/dto/UserInfoResponse.java
package com.xxgwy.modules.user.dto;

import com.xxgwy.modules.user.entity.User;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class UserInfoResponse {
    private UUID id;
    private String username;
    private String nickname;
    private String avatar;
    private String role;

    public static UserInfoResponse from(User user) {
        return new UserInfoResponse(
                user.getId(),
                user.getUsername(),
                user.getNickname(),
                user.getAvatar(),
                user.getRole() != null ? user.getRole().name() : null
        );
    }
}

File: backend/src/main/kotlin/com/xxgwy/modules/user/dto/UserInfoResponse.kt
package com.xxgwy.modules.user.dto

import com.xxgwy.modules.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema
import java.util.UUID

data class UserInfoResponse(
    @Schema(nullable = true)
    val id: UUID?,
    val username: String,
    val nickname: String,
    @Schema(nullable = true)
    val avatar: String?,
    val role: String
) {
    companion object {
        fun from(user: User): UserInfoResponse = UserInfoResponse(
            id = user.id,
            username = user.username,
            nickname = user.nickname,
            avatar = user.avatar,
            role = user.role.name
        )
    }
}
